cpp/src/toppra/geometric_path.cpp

```diff
@@ -10,4 +13,50 @@ Vectors GeometricPath::eval(const Vector &positions, int order) const {
     }
     return outputs;
   };
+
+
+Vector GeometricPath::proposeGridpoints(double max_err_threshold, int max_iteration, double max_seg_length, int min_nb_points) const {
+  std::forward_list<value_type> gridpoints {pathInterval()[0], pathInterval()[1]};
+  int nb_gridpoints = 2;
+
+  // Add points according to error threshold
+  for (auto iter=0; iter < max_iteration; iter++){
+    bool add_new_points = false;
+    for (auto point = gridpoints.begin(); std::next(point, 1) != gridpoints.end(); point++){
+
+      value_type p_mid = 0.5 * (*point + *std::next(point, 1));
+      auto dist = (*std::next(point, 1) - *point);
+
+      if (dist > max_seg_length){
+        gridpoints.emplace_after(point, p_mid);
+        add_new_points = true;
+        nb_gridpoints ++;
+        continue;
+      }
+
+      auto max_err = (0.5 * eval_single(p_mid, 2) * dist * dist).cwiseAbs().maxCoeff();
```

Comment:
Maybe you can comment what this line does. I don't understand it.

Reply:
It's the maximum interpolation error.

Reply:
Hi @jmirabel I have updated the docstring. Hope it's clearer now.